app/helpers/automated_tests_client_helper.rb

```diff
@@ -428,24 +392,121 @@ def self.perform(host_with_port, test_scripts, user_api_key, server_api_key, gro
     end
   end
 
-  def self.process_test_result(raw_result, test_scripts_ran, assignment, grouping, submission, requested_by)
+  def self.process_test_result(xml, test_script_result)

```

Comment:
This should also be an instance method, something like `TestScriptResult#build_test_result_from_xml`.